jupyterhub_moss/spawner.py

```diff
@@ -158,6 +161,8 @@ def create_options_form(spawner):
         "^(?P<hours>[0-9]+)(?::(?P<minutes>[0-5]?[0-9]))?(?::(?P<seconds>[0-5]?[0-9]))?$"
     )
 
+    _MEM_REGEXP = re.compile("^[0-9]*([0-9]+[KMGT])?$")
```

Comment:
This disallows decimal values (such 0.5). Is this expected ?

Also, if I understood correctly, you can only get `<number>G` values from the form so why check for other prefixes ?

Reply:
Yes, there is no decimal in slurm's `--mem` arg.

You can get `0` or `<number>G` from the front-end, but I made it match slurm's `--mem` arg, so there is nothing special to explain for the "spawning through URL".

Reply:
Oh, hadn't thought about the URL spawn. Nice !